@here/harp-examples/src/threejs_add-object.ts

```diff
@@ -40,14 +40,34 @@ export namespace ThreejsAddSimpleObject {
     // snippet:harp_gl_threejs_add_simple_object_0.ts
     const scale = 100;
     const geometry = new THREE.BoxGeometry(1 * scale, 1 * scale, 1 * scale);
+    const prePassMaterial = new THREE.MeshStandardMaterial({
+        color: 0x00ff00fe,
+        opacity: 0.5,
+        depthTest: false,
+        transparent: true
+    });
     const material = new THREE.MeshStandardMaterial({
-        color: 0x00ff00fe
+        color: 0x00ff00fe,
+        opacity: 1.9,
+        transparent: true
     });
-    function createPinkCube(): MapAnchor<THREE.Mesh> {
-        const mesh = new THREE.Mesh(geometry, material);
-        // Make sure the cube overlaps everything else, is completely arbitrary.
+    function createPinkCube(): MapAnchor<THREE.Object3D> {
+        // To avoid not seeing the cube at all if it is fully behind the buildings
+        // and also to have some nice visuals if it is partially occluded we
+        // render two passes:
+        // 1. render the cube semi-transparent w/o depth test (renders entire cube)
+        // 2. render the cube almost opaque w/ depth test (renders only un-occluded part)
+        const cube = new THREE.Object3D();
+
+        const prePassMesh = new THREE.Mesh(geometry, prePassMaterial);
+        prePassMesh.renderOrder = Number.MAX_SAFE_INTEGER - 1;
+        cube.add(prePassMesh);
+
+        const mesh = prePassMesh.clone();
         mesh.renderOrder = Number.MAX_SAFE_INTEGER;
-        return mesh;
+        mesh.material = material;
```

Comment:
does clone not already do this?

Reply:
no, geometry and material are reused when cloning a mesh.

Reply:
yes of course, you are changing the material....😅

Reply:
in saying that, you could avoid 2 lines and just use new THREE.MESH(geometry, material), would save a line

Reply:
ah, true. Will do